### ISSUE TITLE
[ENH] mtype pandas series for univariate forecasting

### ISSUE THREAD

**Issue by kdekker-private** [sktime/sktime#6917]
**Is your feature request related to a problem? Please describe.**
This issue focusses on the pandas allowed mtypes.
It can be good to allow and enforce pandas series mtype for all kinds of univariate forecasting.

Now for series forecasting a pandas dataframe and series is allowed.
For hierachical only a dataframe is allowed.
For univariate forecasting it could make sense to only allow pandas series. Especially, because pandas itself often casts pandas dataframes to pandas series if it notices that it only as one column.
Now the behavior of pandas causes a lot of boilerplate code to enforce data into the right mtype before it enters the forecaster.

**Describe the solution you'd like**
Allow the mtype "pd.series","pd_multiindex_series","pd_multiindex_hier_series" for series, panel and hierarchical forecasting respectively.

**Describe alternatives you've considered**
Boilerplate casting from pandas series to dataframe and back.

**Comment by fkiraly** [comment]
If I understand this correctly (do I?), and you are requesting a *restriction* of the API rather than an extension: This is a "won't do", at least with the current request, because:

* this would be a severe breaking change to the public API
* that would make (single) time series type inconsistent, depending on whether they are univariate or strictly multivariate.

The proposed state for single series was actually the initial point of the API, and we noticed that this type inconsistency required so much case distinctions in the boilerplate, it wasn't good.

It is useful in many places that there is a way to cast all input data to `pd.DataFrame`.

Can you explain where the benefits would lie of doing this?

**Comment by kdekker-private** [comment]
Making the user expectation intuitive is the main aim. Internally under the hood I think it is fine to cast things to pd.dataframe. You already use _check_X_y function for casting. So for the user you ask pd.Series as input, as that is the most intuitive for single column and pd.dataframe is casted to internally.

**Comment by fkiraly** [comment]
the user can pass both - and I would contend there are different, competing user expectations in play.

* `pd.Series` is expected by a user who focuses mostly on univariate cases, because `pd.DataFrame` comes with a slightly unpleasant overhead of handling a single column.
* consistent type is expected by a user who deals with multivariate *and univariate cases simultaneously, the type that satisfies that is `pd.DataFrame`. This type of user would be confused, perhaps annoyed, if we would communicate an expectation that univariate series should be passed as `pd.Series`, but series with two or more columns should be passed as `pd.dataFrame`.

**Comment by fkiraly** [comment]
Following up on this, if I read between the lines: are you suggesting to allow `pd.Series` as mtype for `Panel` and `Hierarchical` as well? As opposed to making a more restrictive recommendation to users?

**Comment by fkiraly** [comment]
ok, this was probably me not reading the original post carefully - you were suggesting to *add* new `pd.Series` based mtypes, as opposed to restricting the API. So your suggestion is simply an API extension, by adding more mtypes. Am I getting this correctly now?

**Comment by kdekker-private** [comment]
Yes you read correctly between the lines as pd.series is more intuitive for univariate cases across the different scitypes.

**Comment by kdekker-private** [comment]
An extension would be good. I can understand that a restriction is not something to do quickly.

**Comment by fkiraly** [comment]
Sure, we can add this!

Adding mtypes is not that intuitive at the moment, but "follow the pattern" should work.

Would you be open to contribute this?

If so, there is also this experimental PR that means to make the mtypes more extensible to users, and enhance documentation on the different mtypes currently supported:

https://github.com/sktime/skpro/pull/392

it's in `skpro` currently, as any negative impacts there are more limited, but ultimately we want to translate this to `sktime`.

**Comment by kdekker-kdr4** [comment]
I am presenting the issue on behalf of my company. Only for critical issues I am allowed to request working hours to help out. It has been decided that this falls not under that. Nonetheless, I will answer any follow up question the best I can.
